clientUtils/formatValue.test.ts

```diff
@@ -0,0 +1,92 @@
+        ["negative quadrillion", -1000000000000000, "-1 quadrillion", {}],
+        ["1000 short prefix", 1000, "1k", { numberAbbreviation: "short" }],
+        ["1499 short prefix", 1499, "1.5k", { numberAbbreviation: "short" }],
+        ["1001 short prefix", 1001, "1k", { numberAbbreviation: "short" }],
+        ["1009 short prefix", 1009, "1.01k", { numberAbbreviation: "short" }],
+        ["12345 short prefix", 12345, "12.35k", { numberAbbreviation: "short" }],
+        ["123456 short prefix", 123456, "123.46k", { numberAbbreviation: "short" }],
+        ["123456 decimal short prefix", 98712.78901, "98.71k", { numberAbbreviation: "short", numDecimalPlaces: 10 }],
+        ["hundred thousand short prefix decimal", 100000.44, "100k", { numberAbbreviation: "short" }],
+        ["1000 long prefix", 1000, "1,000", { numberAbbreviation: "long" }],
+        ["1499 long prefix", 1499, "1,499", { numberAbbreviation: "long" }],
+        ["1001 long prefix", 1001, "1,001", { numberAbbreviation: "long" }],
+        ["1009 long prefix", 1009, "1,009", { numberAbbreviation: "long" }],
+        ["ten thousand long prefix", 10000, "10,000", { numberAbbreviation: "long" }],
+        ["hundred thousand long prefix", 100000, "100,000", { numberAbbreviation: "long" }],
+        ["hundred thousand long prefix decimal", 100000.44, "100,000.44", { numberAbbreviation: "long" }],
+        ["million short prefix", 1000000, "1M", { numberAbbreviation: "short" }],
+        ["billion short prefix", 1000000000, "1B", { numberAbbreviation: "short" }],
+        ["trillion short prefix", 1000000000000, "1T", { numberAbbreviation: "short" }],
+        ["quadrillion short prefix", 1000000000000000, "1quad", { numberAbbreviation: "short" }],
+        ["2 decimals with integer", 1, "1", { numDecimalPlaces: 2 }],
+        ["2 decimals with float", 1.123, "1.12", { numDecimalPlaces: 2 }],
+        ["4 decimals with float", 1.123, "1.123", { numDecimalPlaces: 4 }],
+        ["0 decimals with abbreviation", 1234567, "1.23 million", { numDecimalPlaces: 0, numberAbbreviation: "long" }],
+        ["1 decimal with abbreviation", 1234567, "1.23 million", { numDecimalPlaces: 1, numberAbbreviation: "long" }],
+        ["1 decimal with short abbreviation", 1234, "1.23k", { numDecimalPlaces: 1, numberAbbreviation: "short" }],
+        ["2 decimal with percentage", 19.985, "19.98%", { numDecimalPlaces: 2, unit: "%" }],
+        ["with unit", 1, "$1", { unit: "$" }],
+        ["with custom unit", 1, "1pp", { unit: "pp", spaceBeforeUnit: false }],
+        ["with custom unit and space", 1, "1 pp", { unit: "pp", spaceBeforeUnit: true }],
+        ["negative with unit", -1, "-$1", { unit: "$" }],
+        ["trailingZeroes true", 1.10, "1.1", { trailingZeroes: false }], 
+        ["trailingZeroes false", 1.10, "1.10", { trailingZeroes: true }], 
+        ["$ spaceBeforeUnit false", 1.1, "$1.1", { spaceBeforeUnit: false, unit: "$" }],
```

Comment:
Please add a test case with `{ unit: "£" }`, because I'm currently unsure if that is handled correctly.